Index: src/main/org/h2/store/WriterThread.java
===================================================================
--- src/main/org/h2/store/WriterThread.java	(revision 1655)
+++ src/main/org/h2/store/WriterThread.java	(working copy)
@@ -7,6 +7,7 @@
 package org.h2.store;
 
 import java.lang.ref.WeakReference;
+import java.security.AccessControlException;
 import java.sql.SQLException;
 import org.h2.constant.SysProperties;
 import org.h2.engine.Constants;
@@ -69,12 +70,16 @@
      * @return the writer thread object
      */
     public static WriterThread create(Database database, int writeDelay) {
-        WriterThread writer = new WriterThread(database, writeDelay);
-        Thread thread = new Thread(writer);
-        thread.setName("H2 Log Writer " + database.getShortName());
-        thread.setDaemon(true);
-        thread.start();
-        return writer;
+        try {
+            WriterThread writer = new WriterThread(database, writeDelay);
+            Thread thread = new Thread(writer);
+            thread.setName("H2 Log Writer " + database.getShortName());
+            thread.setDaemon(true);
+            thread.start();
+            return writer;
+        } catch (AccessControlException e) { // Google App Engine does not allow threads
+            return null;
+        }
     }
 
     private LogSystem getLog() {
